arrow/arrow.py

```diff
@@ -352,10 +351,10 @@ def interval(cls, frame, start, end, interval=1, tz=None):
         if interval < 1:
             raise ValueError("interval has to be a positive integer")
 
-        spanRange = cls.span_range(frame,start,end,tz)
+        spanRange = list(cls.span_range(frame, start, end, tz))
 
         bound = (len(spanRange) // interval) * interval
-        return [ (spanRange[i][0],spanRange[i+ interval - 1][1]) for i in range(0,bound, interval) ]
+        return ((spanRange[i][0], spanRange[i + interval - 1][1]) for i in range(0, bound, interval))
```

Comment:
You could keep `spanRange` a generator, and return a generator, with something like this (completely untested, likely missing some boundary cases):

```python
spanRange = cls.span_range(frame, start, end, tz)
while True:
    intvlStart, intvlEnd = next(spanRange)  # StopIteration when exhausted
    for _ in range(interval-1):
        _, intvlEnd = next(spanRange)  # StopIteration when exhausted
    yield intvlStart, intvlEnd
```